zenoh/src/api/handlers/fifo.rs

```diff
@@ -41,11 +49,284 @@ impl Default for FifoChannel {
     }
 }
 
+/// [`FifoChannel`] handler.
+#[derive(Debug, Clone)]
+pub struct FifoChannelHandler<T>(flume::Receiver<T>);
+
 impl<T: Send + 'static> IntoHandler<T> for FifoChannel {
-    type Handler = flume::Receiver<T>;
+    type Handler = FifoChannelHandler<T>;
 
     fn into_handler(self) -> (Callback<T>, Self::Handler) {
-        flume::bounded(self.capacity).into_handler()
+        let (sender, receiver) = flume::bounded(self.capacity);
+        (
+            Callback::new(Arc::new(move |t| {
+                if let Err(error) = sender.send(t) {
+                    tracing::error!(%error)
+                }
+            })),
+            FifoChannelHandler(receiver),
+        )
+    }
+}
+
+impl<T> FifoChannelHandler<T> {
```

Comment:
Shouldn't we mark all those functions #[inline] ?

Reply:
I think that Rust (or LLVM) functions rarely _should_ be `#[inline]`. To quote the [reference](https://doc.rust-lang.org/reference/attributes/codegen.html#the-inline-attribute):

> Note: The rustc compiler automatically inlines functions based on internal heuristics. Incorrectly inlining functions can make the program slower, so this attribute should be used with care.